### PR TITLE
[scratchpad] benchmark tool

### PR DESCRIPTION
**Pull request by lightmark**
## Motivation

We need a tool to bench the scratchpad.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

 cargo run -- --db-dir=/tmp/test --num-updates=10000

### CODE REVIEW

File: storage/scratchpad-benchmark/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        .map(|_| {
+            (
+                AccountAddress::random(),
+                gen_random_blob(blob_size, &mut rng),
+            )
+        })
+        .collect::<Vec<_>>();
+
+    let version = db.get_latest_version().unwrap();
+    let account_state_proofs = updates
```
Comment (msmouse) [inline storage/scratchpad-benchmark/src/lib.rs:48]: Most will be non-existent proofs? 

I think an executor-benchmark run with a lot of accounts created will be a better simulation of the real life situation.

Comment (lightmark) [reply storage/scratchpad-benchmark/src/lib.rs:48]: I can create with a ratio of existent proofs. 
Executor-benchmark uses storage-service which is hard to measure the perf of smt only. Also, it cannot create a super large db in a short time.

Comment (msmouse) [reply storage/scratchpad-benchmark/src/lib.rs:48]: The ProofReader is actually in-mem reads all the time even in the executor-benchmark, so it should be good?

Comment (lightmark) [reply storage/scratchpad-benchmark/src/lib.rs:48]: It is indeed. But for executor benchmark if you want to create a 1M account db you need to wait for 1h?

Comment (lightmark) [reply storage/scratchpad-benchmark/src/lib.rs:48]: executor-benchmark is a superset with some execution that we don't want to measure. I don't see why it is better for this purpose besides it has more existent proofs.

Comment (msmouse) [reply storage/scratchpad-benchmark/src/lib.rs:48]: But 1M accounts is the scenario where we really want to know the improvements. It'll be nice if we can persist the DB with the 1M accounts and share it across runs.

**Review by msmouse — APPROVED** [review]

**Pull request closed without merging**